Competition2018/src/main/java/competition/subsystems/drive/commands/DriveForDistanceCommand.java

```diff
@@ -0,0 +1,111 @@
+    private double targetDistance;
+    private double previousPositionInches;
+    public final double defaultPValue = 1/80d;
+    
+    private DoubleProperty deltaDistanceProp;
+    private ContiguousHeading targetHeading;
+    
+    @Inject
+    public DriveForDistanceCommand(
+            DriveSubsystem driveSubsystem,
+            XPropertyManager propManager,
+            RobotAssertionManager assertionManager,
+            PIDFactory pidFactory,
+            PoseSubsystem pose) {
+        
+    	this.drive = driveSubsystem;
+        this.poseSubsystem = pose;
+        this.requires(driveSubsystem);
+        this.travelManager = pidFactory.createPIDManager("Drive to position", 0.1, 0, 0, 0, 0.5, -0.5, 3, 1, 0.5);
+        headingDrivePid = pidFactory.createPIDManager("Heading module", defaultPValue, 0, 0);
+        targetHeading = new ContiguousHeading();
+    }
+    
+    /**
+     * Sets the target distance
+     * @param deltaDistance the target in inches
+     */
+    public void setDeltaDistance(double deltaDistance) {
+        this.deltaDistance = deltaDistance;
+    }
+    
+    /**
+     * Sets the target distance with a DoubleProperty
+     * @param deltaDistanceProp the DoubleProperty for the target
+     */
+    public void setDeltaDistance(DoubleProperty deltaDistanceProp) {
+        this.deltaDistanceProp = deltaDistanceProp;
+    }   
+    
+    @Override
+    public void initialize() {
+        previousPositionInches = getYDistance();
+        
+        targetHeading = poseSubsystem.getCurrentHeading();
+        
+        if (deltaDistanceProp != null) {
+            this.targetDistance = getYDistance() + deltaDistanceProp.get();
+        } else {
+            this.targetDistance = getYDistance() + deltaDistance;
+        }
+        log.info("Initializing  with distance " + targetDistance + " inches");
+    }
+
+    @Override
+    public void execute() {
+        double power = travelManager.calculate(targetDistance, getYDistance());
+        
+        leftPower = power - calculateHeadingPower();
+        rightPower = power + calculateHeadingPower();
+        
+        drive.drive(leftPower, rightPower);
+    }
+    
+    public double calculateHeadingPower() {
+        double errorInDegrees = targetHeading.difference(poseSubsystem.getCurrentHeading());
+        double normalizedError = errorInDegrees / 180;
+        double rotationalPower = headingDrivePid.calculate(0, normalizedError);
+
+        return rotationalPower;
+    }
+    
+    private double getYDistance() {
+        return poseSubsystem.getRobotOrientedTotalDistanceTraveled().y;
+    }
+    
+    @Override
+    public boolean isFinished() {
+        return  travelManager.isOnTarget();
+    }
+    
+    @Override
+    public void end() {
+        log.info("Ending, PreviousPosition was " + previousPositionInches 
+                + ", Targeted Delta in distance is " + deltaDistance + " Distance traveled is " 
+                + getYDistance());
```

Comment:
Won't `getYDistance` return the total distance travelled since boot rather than just the distance since load?